servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpRequestMetaData.java

```diff
@@ -433,6 +433,33 @@ private void query(final Map<String, List<String>> params) {
         requestTarget(sb.toString());
     }
 
+    @Nullable
+    @Override
+    public String fragment() {
+      return lazyParseRequestTarget().fragment();
+    }
+
+    @Override
+    public HttpRequestMetaData fragment(@Nullable String fragment) {
+
+      if (fragment == null && fragment() == null) {
+        return this;
+      }
+
+      String originalRequestTarget = requestTarget();
+
+      if (lazyParseRequestTarget().fragment() != null) {
+        int fragmentLength = lazyParseRequestTarget().fragment().length() + 1;
+        // remove the existing fragment
+        originalRequestTarget = originalRequestTarget.substring(0, originalRequestTarget.length() - fragmentLength);
+      }
+
+      return requestTarget(fragment == null || fragment.isEmpty()
+              ? originalRequestTarget
```

Comment:
I tested how the `lazyParseRequestTarget()` works, it actually results in the following:
1. `/path` -> fragment() returns `null`
2. `/path#` -> fragment() returns an empty string
3. `/path#foo` -> fragment returns `foo`

Will be great if the setter works in the opposite way: `fragment("")` results in `/path#`, etc.